### PR TITLE
:recycle: Send the monster reference to the monster AI functions

### PR DESCRIPTION
**Pull request by canassa**
These were the last functions still using monsterId on monsters.cpp 🥳

### CODE REVIEW

File: Source/monster.cpp [hunk 1]
```diff
@@ -2343,10 +2316,10 @@ void RhinoAi(int monsterId)
 		if (distanceToEnemy >= 5
 		    && v < 2 * monster.intelligence + 43
 		    && LineClear([&monster](Point position) { return IsTileAvailable(monster, position); }, monster.position.tile, monster.enemyPosition)) {
-			if (AddMissile(monster.position.tile, monster.enemyPosition, md, MIS_RHINO, TARGET_PLAYERS, monsterId, 0, 0) != nullptr) {
+			if (AddMissile(monster.position.tile, monster.enemyPosition, md, MIS_RHINO, TARGET_PLAYERS, monster.getId(), 0, 0) != nullptr) {
 				if (monster.data().hasSpecialSound)
 					PlayEffect(monster, 3);
-				dMonster[monster.position.tile.x][monster.position.tile.y] = -(monsterId + 1);
+				dMonster[monster.position.tile.x][monster.position.tile.y] = -(monster.getId() + 1);
```
Comment (AJenbo) [inline Source/monster.cpp:2322]: `size_t monsterId = monster.getId();` and reuse.

File: Source/monster.cpp [hunk 2]
```diff
@@ -2521,8 +2485,8 @@ void BatAi(int monsterId)
 	    && distanceToEnemy >= 5
 	    && v < 4 * monster.intelligence + 33
 	    && LineClear([&monster](Point position) { return IsTileAvailable(monster, position); }, monster.position.tile, monster.enemyPosition)) {
-		if (AddMissile(monster.position.tile, monster.enemyPosition, md, MIS_RHINO, TARGET_PLAYERS, monsterId, 0, 0) != nullptr) {
-			dMonster[monster.position.tile.x][monster.position.tile.y] = -(monsterId + 1);
+		if (AddMissile(monster.position.tile, monster.enemyPosition, md, MIS_RHINO, TARGET_PLAYERS, monster.getId(), 0, 0) != nullptr) {
+			dMonster[monster.position.tile.x][monster.position.tile.y] = -(monster.getId() + 1);
```
Comment (AJenbo) [inline Source/monster.cpp:2489]: reuse

File: Source/monster.cpp [hunk 3]
```diff
@@ -2753,9 +2699,9 @@ void SnakeAi(int monsterId)
 	unsigned distanceToEnemy = monster.distanceToEnemy();
 	if (distanceToEnemy >= 2) {
 		if (distanceToEnemy < 3 && LineClear([&monster](Point position) { return IsTileAvailable(monster, position); }, monster.position.tile, monster.enemyPosition) && static_cast<MonsterMode>(monster.var1) != MonsterMode::Charge) {
-			if (AddMissile(monster.position.tile, monster.enemyPosition, md, MIS_RHINO, TARGET_PLAYERS, monsterId, 0, 0) != nullptr) {
+			if (AddMissile(monster.position.tile, monster.enemyPosition, md, MIS_RHINO, TARGET_PLAYERS, monster.getId(), 0, 0) != nullptr) {
 				PlayEffect(monster, 0);
-				dMonster[monster.position.tile.x][monster.position.tile.y] = -(monsterId + 1);
+				dMonster[monster.position.tile.x][monster.position.tile.y] = -(monster.getId() + 1);
```
Comment (AJenbo) [inline Source/monster.cpp:2704]: reuse

File: Source/monster.cpp [hunk 4]
```diff
@@ -2849,8 +2792,8 @@ void CounselorAi(int monsterId)
 			} else if (static_cast<MonsterMode>(monster.var1) == MonsterMode::Delay
 			    || GenerateRnd(100) < 2 * monster.intelligence + 20) {
 				StartRangedAttack(monster, MIS_NULL, 0);
-				AddMissile(monster.position.tile, { 0, 0 }, monster.direction, MIS_FLASH, TARGET_PLAYERS, monsterId, 4, 0);
-				AddMissile(monster.position.tile, { 0, 0 }, monster.direction, MIS_FLASH2, TARGET_PLAYERS, monsterId, 4, 0);
+				AddMissile(monster.position.tile, { 0, 0 }, monster.direction, MIS_FLASH, TARGET_PLAYERS, monster.getId(), 4, 0);
+				AddMissile(monster.position.tile, { 0, 0 }, monster.direction, MIS_FLASH2, TARGET_PLAYERS, monster.getId(), 4, 0);
```
Comment (AJenbo) [inline Source/monster.cpp:2796]: reuse

Comment (canassa) [reply Source/monster.cpp]: done

**Review by AJenbo — APPROVED** [review]

**Pull request merged**